src/Client.ts

```diff
@@ -125,20 +127,25 @@ export default class Client extends discord.Client {
     }
 
     const fullPermissions: GuildApplicationCommandPermissionData[] =
-      generatePermissionData(pushedCommands, this.commands.array());
+      generatePermissionData(pushedCommands, [...this.commands.values()]);
 
     // Apply Permissions (per-guild-only)
     await guild.commands.permissions.set({ fullPermissions });
   }
 
+  registerMessageFilters(filters: MessageFilter[]): void {
+    this.messageFilters.push(...filters);
+    this.on('messageCreate', async (message) => await runMessageFilters(message, this.messageFilters));
```

Comment:
You can use `this.messageFilters.forEach()` here to loop over all of the filters.

Reply:
The loop needs to be run sequentially, if a forEach() is used it does't run in sequence.

Reply:
Hmm so is your concern that the next iteration of the loop will start while the deletion is still happening?

Reply:
Yes it causes a race condition, hence the for...of loop

Reply:
Ok I didn't realize this. I think it is good to merge then.